numpy/core/_add_newdocs_scalars.py

```diff
@@ -50,24 +51,44 @@ def type_aliases_gen():
     ])
 
 
+def _get_platform_and_machine():
+    try:
+        system, _, _, _, machine = os.uname()
+    except AttributeError:
+        system = sys.platform
+        if system == 'win32':
+            machine = os.environ.get('PROCESSOR_ARCHITEW6432', '') \
+                    or os.environ.get('PROCESSOR_ARCHITECTURE', '')
+        else:
+            machine = 'unknown'
+    return system, machine
+
+
+_system, _machine = _get_platform_and_machine()
+_doc_alias_string = f":Alias on this platform ({_system} {_machine}):"
```

Comment:
Would it make sense to move this from global scope into the `add_newdoc_for_scalar_type` function?

Reply:
@mattip I placed this outside the `add_newdoc_for_scalar_type` to avoid calling the method for each invocation of `add_newdoc_for_scalar_type`. All objects start with `_`, so it is not part of the public interface.

What I can do is create a single method 
```
@functools.lru_cache(maxsize=None)
def _get_doc_alias_string():
   # returns a string of the form ":Alias on this platform ({_system} {_machine}):"
   ...
```
and then do 
```
def add_newdoc_for_scalar_type(obj, fixed_aliases, doc):
    _doc_alias_string = __get_doc_alias_string()
   ...
```
(The added time of `functools.lru_cache` for each invocation is nearly zero).

Reply:
That seems too complicated, let's go with what you have.